File: DeltaX.DemoServerTodoAppV1/Dtos/TodoDto.cs
namespace DeltaX.DemoServerTodoAppV1.Dtos
{
    using System;

    public record TodoDto
    {
        public int Id { get; set; }
        public string Description { get; set; }
        public bool Completed { get; set; }
        public DateTimeOffset Created { get; set; }
        public DateTimeOffset Updated { get; set; }
    }
}

File: DeltaX.DemoClientVueTodoAppV1/src/dtos/TodoDto.ts
export interface TodoDto {
    id: number;
    description: string;
    completed: boolean;
    created: Date;
    updated: Date;
}

export interface TodoUpdateDto {
    id: number;
    description: string;
    completed: boolean;
}

export interface TodoCreateDto {
    description: string;
    completed: boolean;
}

export interface TodoDtoDataTrackerResultDto {
    items?: { updated: Date, reason: number, item: TodoDto }[];
    first?: Date;
    last?: Date;
}

export interface PaginatingTodosDto {
    items: TodoDto[];
    maxResultCount: number;
    skipCount: number;
    totalCount: number;
}